tinygrad/runtime/metal.py

```diff
@@ -22,6 +22,8 @@ def __init__(self, shape): raise NotImplementedError("Metal runtime doesn't supp
 
 class CLBuffer:
   def __init__(self, size): self._cl = device.newBufferWithLength_options_(size, Metal.MTLResourceStorageModeShared)
+  def __del__(self): self._cl.release()
```

Comment:
This doesn't happen by default on out of scope?

Reply:
didnt dive deep into it, but noticed that test_speed_v_torch execution time reduces after adding that
without release ~24 seconds
with release ~19 seconds

Reply:
Repeatability? I can't see why this would have an affect.

Reply:
yes, just ran some tests now and watched the memory usage and without releasing it kept growing

Reply:
Which raises deeper questions about pyobic objects...will merge